payments-core/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -190,6 +182,47 @@ internal class PaymentSheetViewModel @Inject internal constructor(
                 stripeIntentValidator.requireValid(stripeIntent)
             }.fold(
                 onSuccess = {
+                    updatePaymentMethods(stripeIntent)
+                    resetViewState(stripeIntent, userErrorMessage = null)
+                },
+                onFailure = ::onFatal
+            )
+        }
+    }
+
+    /**
+     * Fetch the saved payment methods for the customer, if a [PaymentSheet.CustomerConfiguration]
+     * was provided.
+     * It will fetch only the payment method types accepted by the [stripeIntent] and defined in
+     * [SupportedSavedPaymentMethod].
+     */
+    @VisibleForTesting
+    fun updatePaymentMethods(stripeIntent: StripeIntent) {
+        viewModelScope.launch {
+            runCatching {
+                customerConfig?.let { customerConfig ->
+                    stripeIntent.paymentMethodTypes.mapNotNull {
+                        SupportedSavedPaymentMethod.fromCode(it)
+                    }.map {
+                        async {
+                            paymentMethodsRepository.get(
+                                customerConfig,
+                                it.type
+                            )
+                        }
+                    }.awaitAll().flatten().filter { paymentMethod ->
```

Comment:
Nice. Should this be moved to `StripeApiRepository`? I could see this being reused.

Reply:
I moved up to PaymentMethodsApiRepository [here](https://github.com/stripe/stripe-android/compare/brnunes/savedpm...brnunes/getpm?expand=1), trying to understand why when the `stripeRepository` call fails the method throws instead of returning the empty list from `getOrDefault`. I'll continue on that next week.

Reply:
sweet!

Reply:
Is moving it to PaymentMethodsApiRepository just an iteration before moving it to StripeApiRepository, or is there a reason you decided against the StripeApiRepository?

I think you can also move the check that verifies the PaymentMethod returned is one of the paymentMethod types requested to either StripeApiRepository or PaymentMethodsApiRepository.

Reply:
I'll move to `StripeApiRepository` on a follow-up PR. The check verifies if the payment method has the content we require in Payment Sheet, I don't think that part should be moved.